brave/src/test/java/brave/features/baggage/DynamicBaggageHandler.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.features.baggage;
+
+import brave.baggage.BaggageField;
+import brave.internal.Nullable;
+import brave.internal.baggage.BaggageHandler;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import org.apache.logging.log4j.core.util.StringBuilderWriter;
+
+import static com.google.common.base.Objects.equal;
+
+/** This accepts any keys, but only writes to one propagation field. */
+final class DynamicBaggageHandler implements BaggageHandler<Map<BaggageField, String>> {
+  @Override public boolean isDynamic() {
+    return true;
+  }
+
+  @Override public List<BaggageField> currentFields(@Nullable Map<BaggageField, String> state) {
+    if (state == null) return Collections.emptyList();
+    return new ArrayList<>(state.keySet());
+  }
+
+  @Override public boolean handlesField(BaggageField field) {
+    return true; // grow indefinitely
+  }
+
+  @Override public String getValue(BaggageField field, Map<BaggageField, String> state) {
+    return state.get(field);
+  }
+
+  @Override public Map<BaggageField, String> newState(BaggageField field, String value) {
+    LinkedHashMap<BaggageField, String> newState = new LinkedHashMap<>();
+    newState.put(field, value);
+    return newState;
+  }
+
+  @Override
+  public Map<BaggageField, String> mergeState(Map<BaggageField, String> state, BaggageField field,
+    String value) {
+    if (equal(value, state.get(field))) return state;
+    if (value == null) {
+      if (!state.containsKey(field)) return state;
+      if (state.size() == 1) return null;
```

Comment:
`Collections.emptyMap`? Otherwise the interface needs to be updated to have a `@Nullable` return with what it means, but empty map seems simpler without any overhead.

Reply:
punted to #1153.. agree empty map doesn't have more overhead as the copy/constructor of LHM is cheap.  However, it does complicate the state model. let me know what you think..